File: daybreak/src/daybreak/adapters/terminal/wrappers.py
from daybreak.terminals.base import Terminal
from daybreak.terminals.universal import UniversalPty


class UniversalPtyAdapter:
    name = "universal_pty"

    def __init__(self):
        self.terminal = UniversalPty()

    def apply_mode(self, mode: str, theme_name: str, palette: dict = None):
        self.terminal.apply_theme(theme_name, mode)


class LegacyModeTerminalAdapter:
    def __init__(self, terminal: Terminal, name: str = None):
        self.terminal = terminal
        self.name = name or terminal.__class__.__name__

    def apply_mode(self, mode: str, theme_name: str, palette: dict = None):
        self.terminal.set_mode(mode)

File: src/adapters/terminal/wrappers.ts
import { Terminal } from '../../terminals/base';
import { UniversalPty } from '../../terminals/universal';
import { Palette } from '../../colors';

export class UniversalPtyAdapter {
  name = 'universal_pty';
  private terminal: UniversalPty;

  constructor() {
    this.terminal = new UniversalPty();
  }

  applyMode(mode: string, themeName: string, _palette?: Palette): void {
    this.terminal.applyTheme(themeName, mode);
  }
}

export class LegacyModeTerminalAdapter {
  name: string;
  private terminal: Terminal;

  constructor(terminal: Terminal, name?: string) {
    this.terminal = terminal;
    this.name = name || terminal.constructor.name;
  }

  applyMode(mode: string, _themeName: string, _palette?: Palette): void {
    this.terminal.setMode(mode);
  }
}
